portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -78,10 +83,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
   public String getParameter(HttpServletRequest req, String key, String defaultVal) {
       String value = req.getParameter(key);
 
-      // Protect against adding null to our comments array.
-      if (value == null) {
+      // Protect against adding null / empty input to our comments array.
+      if (value == null || value.equals("")) {
           return defaultVal;
       }
       return value;
   }
+
+  private static class Comment {
```

Comment:
I think it is better to put definition of Comment class before its first use. This way, when reading the class top to bottom, the reader would be aware of comment fields before their first use.